Description: Fix the build with gcc 4.7.0
 .
 The part in Source/common.pri has been adapted to use the version of g++
 used during build, not the one used to build Qt4.
Author: Andras Becsi <andras.becsi@nokia.com>
Author: Pino Toscano <pino@debian.org>
Bug: https://bugs.webkit.org/show_bug.cgi?id=83584
Origin: upstream, http://trac.webkit.org/changeset/113848

---
 Source/WebCore/html/HTMLImageElement.cpp                   |    2 +-
 Source/WebCore/platform/graphics/TiledBackingStoreClient.h |    1 +
 Source/common.pri                                          |    5 +++--
 3 files changed, 5 insertions(+), 3 deletions(-)

--- a/Source/WebCore/html/HTMLImageElement.cpp
+++ b/Source/WebCore/html/HTMLImageElement.cpp
@@ -74,7 +74,7 @@ PassRefPtr<HTMLImageElement> HTMLImageEl
     RefPtr<HTMLImageElement> image = adoptRef(new HTMLImageElement(imgTag, document));
     if (optionalWidth)
         image->setWidth(*optionalWidth);
-    if (optionalHeight > 0)
+    if (optionalHeight)
         image->setHeight(*optionalHeight);
     return image.release();
 }
--- a/Source/WebCore/platform/graphics/TiledBackingStoreClient.h
+++ b/Source/WebCore/platform/graphics/TiledBackingStoreClient.h
@@ -25,6 +25,7 @@ namespace WebCore {
 #if ENABLE(TILED_BACKING_STORE)
 class TiledBackingStoreClient {
 public:
+    virtual ~TiledBackingStoreClient() { }
     virtual void tiledBackingStorePaintBegin() = 0;
     virtual void tiledBackingStorePaint(GraphicsContext*, const IntRect&) = 0;
     virtual void tiledBackingStorePaintEnd(const Vector<IntRect>& paintedArea) = 0;
--- a/Source/common.pri
+++ b/Source/common.pri
@@ -5,8 +5,9 @@ contains(JAVASCRIPTCORE_JIT,no): DEFINES
 
 linux-g++ {
 isEmpty($$(SBOX_DPKG_INST_ARCH)):exists(/usr/bin/ld.gold) {
-    message(Using gold linker)
-    QMAKE_LFLAGS+=-fuse-ld=gold
+    GCC_VERSION_MINOR = $$system($$QMAKE_CXX -dumpversion | sed -r -e 's,\\([0-9]+\\)\\\\.\\([0-9]+\\).*,\\\\2,g')
+    # Upstream gcc 4.7 does not support the -fuse-ld=gold option ( see: http://sourceware.org/ml/binutils/2011-01/msg00178.html )
+    isEqual(QT_GCC_MAJOR_VERSION, 4):lessThan(GCC_VERSION_MINOR, 7):QMAKE_LFLAGS+=-fuse-ld=gold
 }
 }
 
